README.md

# Project not mainained any more
This project is not maintained anymore, please use https://extensions.typo3.org/extension/rx_shariff/ as replacement.
This project will be removed in some month.


ChangeLog
09-09-11 Frank Nägler <typo3@naegler.net>
	* release 2.0.7
	* small changes to make this extension compatible with 6.2.x

09-09-11 Frank Nägler <typo3@naegler.net>
	* release 2.0.6
	* service change: #0000046: hype (http://hype.yeebase.com) was renamed to t3n socialnews (http://t3n.de/socialnews/), so submit url and icon changed
	* performance improvment
	* service change: #0000037: new icon for mr. wong

07-12-08 Frank Nägler <typo3@naegler.net>
	* release 2.0.3
	* bugfix: wrong call for stdWrap on iconWrap
	* added new bookmark service (hype.yeebase.com) hype it! yeah ;-)

07-11-05 Frank Nägler <typo3@naegler.net>
	* release 2.0.2
	* bugfix: wrong TypoScript config #0000030

07-11-03 Frank Nägler <typo3@naegler.net>
	* release 2.0.1
	* bugfix: wrong TypoScript config (copy failure)

07-10-13 Frank Nägler <typo3@naegler.net>
	* release 2.0.0
	* reimplementation of the extension.
	* services can be configured by TypoScript
	* documentation updated
	* added new bookmark service (oneview.de) thanks to Christoph Moeller for this hint

07-10-05 Frank Nägler  <typo3@naegler.net>
	* added new bookmark service (tagthat.de) thanks to Markus Gevers for this hint
	* bugfix #0000028: .tagline has no effect

07-07-11 Frank Nägler  <typo3@naegler.net>
	* improvement of the generated URLs (& = &amp;)
	* added new bookmark service (folkd.com) thanks to Bernd Wilke for this hint
	* added new bookmark service (linkarena.com) thanks to Bernd Wilke for this hint
	* added new bookmark service (webnews.de) thanks to Bernd Wilke for this hint

07-01-23 Frank Nägler  <typo3@naegler.net>
	* added new bookmark service (yigg.de)

06-12-27 Frank Nägler  <typo3@naegler.net>
    * added new bookmark service (addthis.com)
    * added new bookmark service (ask.com)
    * added new bookmark service (backflip.com)
    * added new bookmark service (stumbleupon.com)
    * added new bookmark service (google.com)
    * added new bookmark service (live.com)
    * added new bookmark service (technorati.com)
    * added new bookmark service (netscape.com)
    * added new bookmark service (wink.com)
    * added new bookmark service (bluedot.us)
    
    
06-12-19 Frank Nägler  <typo3@naegler.net>
    * new feature: know you can use this extension on every page not only on Blog-Posts
    
06-10-06 Frank Nägler  <typo3@naegler.net>
    * added new bookmark service (de.lirio.us)
    * added new bookmark service (fark.com)
    * added new bookmark service (feedmelinks.com)
    * added new bookmark service (linkagogo.com)
    * added new bookmark service (netvouz.com)
    * added new bookmark service (rawsugar.com)
    * added new bookmark service (scuttle.org)
    * added new bookmark service (shadows.com)
    * added new bookmark service (simpy.com)
    * added new bookmark service (smarking.com)
    * added new bookmark service (wists.com)
    * bugfix #0000011: Path to image incorrect

06-09-10 Frank Nägler  <typo3@naegler.net>
    * added new bookmark service (digg.com)
    * added new bookmark service (newsvine.com)

06-09-09 Frank Nägler  <typo3@naegler.net>

    * added documentation
    * added new social bookmark service (mylink.de)

06-09-08 Frank Nägler  <typo3@naegler.net>

    * Initial release


pi1/class.tx_timtabsociable_pi1.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2006 Frank Nägler <typo3@naegler.net>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/

if (!class_exists('tslib_pibase')) require_once(PATH_tslib . 'class.tslib_pibase.php');


/**
 * Plugin 'social bookmark icons' for the 'timtab_sociable' extension.
 *
 * @author	Frank Nägler <typo3@naegler.net>
 * @package	TYPO3
 * @subpackage	tx_timtabsociable
 */
class tx_timtabsociable_pi1 extends tslib_pibase {
	var $prefixId = 'tx_timtabsociable_pi1';		// Same as class name
	var $scriptRelPath = 'pi1/class.tx_timtabsociable_pi1.php';	// Path to this script relative to the extension dir.
	var $extKey = 'timtab_sociable';	// The extension key.
	var $pi_checkCHash = TRUE;

	/**
	 * The main method of the PlugIn
	 *
	 * @param	string		$content: The PlugIn content
	 * @param	array		$conf: The PlugIn configuration
	 * @return	The content that is displayed on the website
	 */
	function main($content,$conf)	{
		$this->conf=$conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();
		$this->getVars = t3lib_div::_GET();
		$this->tt_newsVars = $this->getVars['tx_ttnews'];
		$this->postID = (isset($this->tt_newsVars['tt_news'])) ? intval($this->tt_newsVars['tt_news']) : (null);
		$this->local_cObj = t3lib_div::makeInstance('tslib_cObj');

		
		$content='';
		$tagline='';
		
		if (isset($this->conf['tagline.'])) $tagline = $this->local_cObj->stdWrap($this->pi_getLL('tagline'),$this->conf['tagline.']);
		$content .= $tagline;

		$activeServices = t3lib_div::trimExplode(',', $this->conf['activeServices'], true);
		for ($i=0;$i<count($activeServices);$i++) {
			$service = $activeServices[$i];
			$target = (strlen($this->conf['services.'][$service.'.']['target'])) ? (' target="'.$this->conf['services.'][$service.'.']['target'].'"') : ('');
			$icon = '<a href="'.$this->conf['services.'][$service.'.']['url'].'" title="'.$this->conf['services.'][$service.'.']['title'].'"'.$target.'>'.$this->local_cObj->IMAGE($this->conf['services.'][$service.'.']['icon.']).'</a>';
			$content .= (isset($this->conf['services.'][$service.'.']['wrap'])) ? ($this->local_cObj->stdWrap($icon, $this->conf['services.'][$service.'.'])) : ($icon);
		} 

       $content = $this->prepareOutput($content);
		if (isset($this->conf['outerWrap'])) $content = $this->local_cObj->stdWrap($content,$this->conf['outerWrap.']);
		return $this->pi_wrapInBaseClass($content);
	}

	/**
    * prepare the output for rendering, replace the placeholder with content
    *
    * @param	string	$cont: content with placeholder
    * @return   string  parsed content with replaces placeholder
    */
	function prepareOutput($cont) {
		if ($this->postID) {
		        $res = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
	        	    'title',
		            'tt_news',
	        	    'uid=' . $this->postID . $this->local_cObj->enableFields('tt_news')
		        );
		        $row = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res);
			$title = $row['title'];
		} else {
			$where = 'uid='.$GLOBALS['TSFE']->id;
			$res = $GLOBALS['TYPO3_DB']->exec_SELECTquery('*', 'pages', $where);
			$row = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($res);
			if (is_array($row) && count($row) > 0) {
				$this->local_cObj->start($row, 'pages');
				$title = $this->local_cObj->cObjGetSingle($this->conf['alternativeTitle'],$this->conf['alternativeTitle.']);
			} 
		}
		$url = t3lib_div::getIndpEnv('TYPO3_REQUEST_URL');
		if ((!isset($url)) || ($url == ''))
			$url = substr(t3lib_div::getIndpEnv('TYPO3_SITE_URL'), 0, strlen(t3lib_div::getIndpEnv('TYPO3_SITE_URL'))-1) . t3lib_div::getIndpEnv('REQUEST_URI');

        $cont = str_replace('%URL%', urlencode($url), $cont);
        $cont = str_replace('%URL_DOUBLE_ENCODED%', urlencode(urlencode($url)), $cont);
        $cont = str_replace('%TITLE%', urlencode($title), $cont);
        return $cont;
    }
}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/timtab_sociable/pi1/class.tx_timtabsociable_pi1.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/timtab_sociable/pi1/class.tx_timtabsociable_pi1.php']);
}

?>


pi1/class.tx_timtabsociable_pi1_wizicon.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2006 Frank Nägler <typo3@naegler.net>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/




/**
 * Class that adds the wizard icon.
 *
 * @author	Frank Nägler <typo3@naegler.net>
 * @package	TYPO3
 * @subpackage	tx_timtabsociable
 */
class tx_timtabsociable_pi1_wizicon {

	/**
	 * Processing the wizard items array
	 *
	 * @param	array		$wizardItems: The wizard items
	 * @return	Modified array with wizard items
	 */
	function proc($wizardItems)	{
		global $LANG;

		$LL = $this->includeLocalLang();

		$wizardItems['plugins_tx_timtabsociable_pi1'] = array(
			'icon'=>t3lib_extMgm::extRelPath('timtab_sociable').'pi1/ce_wiz.gif',
			'title'=>$LANG->getLLL('pi1_title',$LL),
			'description'=>$LANG->getLLL('pi1_plus_wiz_description',$LL),
			'params'=>'&defVals[tt_content][CType]=list&defVals[tt_content][list_type]=timtab_sociable_pi1'
		);

		return $wizardItems;
	}

	/**
	 * Reads the [extDir]/locallang.xml and returns the $LOCAL_LANG array found in that file.
	 *
	 * @return    The        array with language labels
	 */
	function includeLocalLang()    {
		$llFile = t3lib_extMgm::extPath('timtab_sociable').'locallang.xml';

		$version =     class_exists('t3lib_utility_VersionNumber')
			? t3lib_utility_VersionNumber::convertVersionNumberToInteger(TYPO3_version)
			: t3lib_div::int_from_ver(TYPO3_version);
		if ($version >= 4007000) {
			$object = t3lib_div::makeInstance('t3lib_l10n_parser_Llxml');
			$LOCAL_LANG =  $object->getParsedData($llFile, $GLOBALS['LANG']->lang);
		} else {
			$LOCAL_LANG =  t3lib_div::readLLXMLfile($llFile, $GLOBALS['LANG']->lang);
		}

		return $LOCAL_LANG;
	}
}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/timtab_sociable/pi1/class.tx_timtabsociable_pi1_wizicon.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/timtab_sociable/pi1/class.tx_timtabsociable_pi1_wizicon.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "timtab_sociable"
#
# Auto generated 08-05-2009 23:09
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'TIMTAB - social bookmark icons',
	'description' => 'this extension shows a line of small icons of social bookmark services. with this buttons an user can add your blogpost easy to the services. Please report bugs, or new feature request at https://forge.typo3.org/projects/extension-timtab_sociable',
	'category' => 'plugin',
	'shy' => 0,
	'version' => '2.0.7',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'stable',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Frank Naegler',
	'author_email' => 'typo3@naegler.net',
	'author_company' => 'http://naegler.hamburg',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'cms' => '',
			'php' => '5.2.0-5.4.99',
			'typo3' => '4.5.0-6.2.99',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:59:{s:9:"ChangeLog";s:4:"e3f4";s:10:"README.txt";s:4:"7efc";s:12:"ext_icon.gif";s:4:"6dc2";s:17:"ext_localconf.php";s:4:"7a10";s:14:"ext_tables.php";s:4:"8ab2";s:13:"locallang.xml";s:4:"2e05";s:16:"locallang_db.xml";s:4:"2ea8";s:14:"doc/manual.sxw";s:4:"1d29";s:14:"pi1/ce_wiz.gif";s:4:"9059";s:35:"pi1/class.tx_timtabsociable_pi1.php";s:4:"95db";s:43:"pi1/class.tx_timtabsociable_pi1_wizicon.php";s:4:"ab75";s:13:"pi1/clear.gif";s:4:"cc11";s:17:"pi1/locallang.xml";s:4:"b786";s:19:"res/img/addthis.gif";s:4:"3955";s:15:"res/img/ask.png";s:4:"bdcc";s:20:"res/img/backflip.png";s:4:"855a";s:21:"res/img/blinkbits.png";s:4:"c871";s:21:"res/img/blinklist.png";s:4:"30d6";s:21:"res/img/blogmarks.png";s:4:"e0ee";s:19:"res/img/bluedot.png";s:4:"0ee6";s:21:"res/img/co.mments.gif";s:4:"3a8a";s:20:"res/img/connotea.png";s:4:"3ca2";s:21:"res/img/delicious.png";s:4:"a067";s:21:"res/img/delirious.png";s:4:"f7f1";s:16:"res/img/digg.gif";s:4:"2544";s:16:"res/img/fark.png";s:4:"1ccb";s:23:"res/img/feedmelinks.png";s:4:"f7a1";s:17:"res/img/folkd.png";s:4:"45be";s:16:"res/img/furl.png";s:4:"d7c6";s:18:"res/img/google.png";s:4:"49df";s:16:"res/img/hype.gif";s:4:"3374";s:21:"res/img/linkagogo.png";s:4:"fbb9";s:21:"res/img/linkarena.png";s:4:"098f";s:16:"res/img/live.gif";s:4:"f5fe";s:20:"res/img/magnolia.png";s:4:"b3f9";s:20:"res/img/mylinkde.png";s:4:"b6f8";s:20:"res/img/netscape.gif";s:4:"64ea";s:19:"res/img/netvouz.png";s:4:"28a7";s:20:"res/img/newsvine.gif";s:4:"39b6";s:19:"res/img/oneview.gif";s:4:"9fca";s:20:"res/img/rawsugar.png";s:4:"8cca";s:18:"res/img/reddit.png";s:4:"bb46";s:19:"res/img/scuttle.png";s:4:"2b89";s:19:"res/img/shadows.png";s:4:"77d9";s:17:"res/img/simpy.png";s:4:"cb48";s:20:"res/img/smarking.png";s:4:"8aff";s:17:"res/img/spurl.png";s:4:"c15c";s:14:"res/img/su.png";s:4:"1ad4";s:19:"res/img/tagthat.gif";s:4:"2e06";s:20:"res/img/tailrank.png";s:4:"2760";s:22:"res/img/technorati.png";s:4:"5cc3";s:19:"res/img/webnews.png";s:4:"c411";s:16:"res/img/wink.png";s:4:"b43c";s:17:"res/img/wists.png";s:4:"1e7e";s:16:"res/img/wong.png";s:4:"d3ed";s:22:"res/img/yahoomyweb.png";s:4:"e2dd";s:18:"res/img/yiggit.png";s:4:"822b";s:36:"static/timtab_sociable/constants.txt";s:4:"5069";s:32:"static/timtab_sociable/setup.txt";s:4:"0d48";}',
);
